kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/internal/expressions/DefaultExpressionEvaluatorSuite.scala

```diff
@@ -0,0 +1,255 @@
+      Literal.ofDate(4234),
+      Literal.ofNull(DateType.INSTANCE),
+      Literal.ofTimestamp(2342342342232L),
+      Literal.ofNull(TimestampType.INSTANCE))
+
+    val inputBatches: Seq[ColumnarBatch] = Seq[ColumnarBatch](
+      emptyColumnarBatch(0),
+      emptyColumnarBatch(25),
+      emptyColumnarBatch(128))
+
+    for (literal <- testLiterals) {
+      val outputDataType = literal.getDataType
+      for (inputBatch <- inputBatches) {
+        val outputVector: ColumnVector =
+          evaluator(inputBatch.getSchema, literal, literal.getDataType)
+            .eval(inputBatch)
+
+        assert(inputBatch.getSize === outputVector.getSize)
+        assert(outputDataType === outputVector.getDataType)
+
+        for (rowId <- 0 until outputVector.getSize) {
+          if (literal.getValue == null) {
+            assert(
+              outputVector.isNullAt(rowId),
+              s"expected a null at $rowId for $literal expression")
+          } else {
+            assert(
+              literal.getValue === getValueAsObject(outputVector, rowId),
+              s"invalid value at $rowId for $literal expression"
+            )
+          }
+        }
+      }
+    }
+  }
+
+  // Literals of complex types are not supported.
+  // TODO: should this be moved to kernel-api module?
+  test("expression: unsupported literal data types") {
+    val ex1 = intercept[IllegalArgumentException] {
+      Literal.ofNull(new ArrayType(IntegerType.INSTANCE, true))
+    }
+    assert(ex1.getMessage.contains("array[integer] is an invalid data type for Literal."))
+
+    val ex2 = intercept[IllegalArgumentException] {
+      Literal.ofNull(new MapType(IntegerType.INSTANCE, IntegerType.INSTANCE, true))
+    }
+    assert(ex2.getMessage.contains("map[integer, integer] is an invalid data type for Literal."))
+
+    val ex3 = intercept[IllegalArgumentException] {
+      Literal.ofNull(new StructType().add("s1", BooleanType.INSTANCE))
+    }
+    assert(ex3.getMessage.matches("struct.* is an invalid data type for Literal."))
+  }
+
+  PRIMITIVE_TYPES.foreach { dataType =>
+    test(s"expression: column of type $dataType") {
+      val batchSize = 78;
+      val batchSchema = new StructType().add("col1", dataType)
+      val batch = new DefaultColumnarBatch(
+        batchSize,
+        batchSchema,
+        Array[ColumnVector](testColumnVector(batchSize, dataType)))
+
+      val outputVector = evaluator(batchSchema, new Column("col1"), dataType)
+        .eval(batch)
+
+      assert(batchSize === outputVector.getSize)
+      assert(dataType === outputVector.getDataType)
+      Seq.range(0, outputVector.getSize).foreach { rowId =>
+        assert(
+          generateIsNull(dataType, rowId) === outputVector.isNullAt(rowId),
+          s"unexpected nullability at $rowId for $dataType type vector")
+        if (!outputVector.isNullAt(rowId)) {
+          assert(
+            generateValue(dataType, rowId) === getValueAsObject(outputVector, rowId),
```

Comment:
nit: its easier to read this if generateValue is renamed to testValue. generateValue does not convey the meaning that the values are consistent. I had to look up the docs to understand. but testValue conveys that, and it consistent with the testColumnVector that is materializing the testValues.